hoomd/simulation.py

```diff
@@ -137,7 +137,11 @@ def create_state_from_snapshot(self, snapshot):
         if self.state is not None:
             raise RuntimeError("Cannot initialize more than once\n")
 
-        self._state = State(self, snapshot)
+        try:
```

Comment:
Some developer comments that state we only support `hoomd.Snapshot` and `gsd.hoomd.Snapshot` would be helpful here.

Reply:
Good point. I'll also modify the docstring.